h/services/annotation_moderation.py

```diff
@@ -23,9 +23,17 @@ def all_hidden(self, annotation_ids: str) -> set[str]:
         return {a.id for a in query if a.is_hidden}
 
     def set_status(
-        self, annotation: Annotation, user: User, status: ModerationStatus | None
+        self,
+        annotation: Annotation,
+        status: ModerationStatus | None,
+        user: User | None = None,
     ) -> None:
         """Set the moderation status for an annotation."""
+
+        if not user:
```

Comment:
I tried the alternative, passing user directly all the way here in all cases but that meant that a lot of interfaces need to change to basically get the annotation author.

I reckon this condition is a good compromise.